common/src/main/java/io/seata/common/util/StringUtils.java

```diff
@@ -349,6 +353,33 @@ public static String hump2Line(String str) {
         return sb.toString();
     }
 
+    /**
+     * check string data size
+     *
+     * @param data the str
+     * @param dataName the data name
+     * @param errorSize throw exception if size > errorSize
+     * @return boolean
+     */
+    public static boolean checkDataSize(String data, String dataName, int errorSize, boolean throwIfErr) {
+        try {
+            if (isBlank(data)) {
+                return true;
+            }
+            int length = data.getBytes(Constants.DEFAULT_CHARSET_NAME).length;
+            if (length > errorSize) {
+                LOGGER.warn("{} data is large(errorSize), size={}", dataName, length);
+                if (!throwIfErr) {
+                    return false;
+                }
+                throw new IllegalArgumentException(dataName + " data is too large, size=" + length);
+            }
+        } catch (UnsupportedEncodingException e) {
+            LOGGER.error("decode name error: {}", e.getMessage(), e);
```

Comment:
1. Can non-user input programs guarantee the correct encoding of names?

2. Is it appropriate for a lower-level utility class to directly catch exceptions and have the ability to handle business exceptions from upper layers?

Reply:
> 1. Can non-user input programs guarantee the correct encoding of names?
> 2. Is it appropriate for a lower-level utility class to directly catch exceptions and have the ability to handle business exceptions from upper layers?

1 . With reference to codec, wouldn't it be better to use utf8 by default? Since this detection behavior is not known to the user so UnsupportedEncodingException can be changed to log.warn after catching it.
2 . I'm only catching UnsupportedEncodingException from encoding here, why this practice is catching upper level business exception?